### PR TITLE
get Radio version for SX1276 and SX1280

### PR DESCRIPTION
**Pull request by Nitish1802**
Get radio version for SX1276 and SX1280 using AT command
Command: AT+RADVER?

Tests Done:
- Compile Successful
- Tested on SX1276, displayed 0x12 default value as per Datasheet.

### CODE REVIEW

File: phy/sx1276/sx1276_src/sx1276.c [hunk 1]
```diff
@@ -459,6 +460,7 @@ void initRadio(void)
     setCodingRate4(2);    
     //125kHz optimization
     version = SX1276Read(REG_LR_DETECTOPTIMIZE);
+    getSX1276_ver = version;
```
Comment (sramrajkar) [inline phy/sx1276/sx1276_src/sx1276.c:463]: Remove this. I have reused the variable. Here we reassign the variable with some different register

File: app/app_src/application.c [hunk 2]
```diff
@@ -65,6 +65,12 @@ static void free_tx_buffer(NWK_DataReq_t *req, bool ack);
 static bool get_free_rx_buffer(uint8_t *buf_id);
 static void exract_sink_addr(uint8_t* dataptr);
 
+#ifdef SX1276
+    uint8_t getSX1276_ver;
+#elif defined SX1280
+    uint16_t getSX1280_ver;
+#endif
```
Comment (sramrajkar) [inline app/app_src/application.c:72]: consider using single variable name for all radios. May be move this to application.h?

File: app/app_src/application.c [hunk 3]
```diff
@@ -971,6 +977,22 @@ static void cmdGetRFCH(){
...
+ * \param [OUT] None.
+ * \param [IN] None.
+ */
+static void cmdGetRadioVer(){
+  
+    #ifdef SX1276
+        printf("RADIO VERSION=%x\r\n",getSX1276_ver);
+    #elif defined SX1280
+        printf("RADIO VERSION=%x\r\n",getSX1280_ver);
+    #endif
```
Comment (sramrajkar) [inline app/app_src/application.c:992]: Can definitely made one common statement, once we use common version variable

File: phy/sx1276/sx1276_src/sx1276.c [hunk 4]
```diff
@@ -442,12 +442,11 @@ uint8_t read()
...
 void initRadio(void)
 {
-    uint8_t version;
     uint16_t wideRSSI;
     //Read the radio version
-    version = SX1276Read(REG_LR_VERSION);
-    version++;
-	(void) version; //Suppress static analysis warning
+    rad_version = SX1276Read(REG_LR_VERSION);
+    rad_version++;
```
Comment (Nitish1802) [inline phy/sx1276/sx1276_src/sx1276.c:448]: I tested for SX1276, now the result shows 0x13, not sure why here the version is incremented.

File: app/app_src/application.c (hunk 3 above)
Comment (Nitish1802) [reply app/app_src/application.c]: used common radio version variable

File: app/app_inc/application.h [hunk 5]
```diff
@@ -89,6 +89,7 @@ enum UART_BAUD_ENUM uart_baud_rate; // Used for MB map
 enum UART_PARITY_ENUM uart_parity;
 uint8_t curent_parity;
 uint16_t pan_id;
+uint16_t rad_version;
```
Comment (sramrajkar) [inline app/app_inc/application.h:92]: I think this can be uint8_t for all radios used so far

File: phy/sx1276/sx1276_src/sx1276.c (hunk 4 above)
Comment (sramrajkar) [inline phy/sx1276/sx1276_src/sx1276.c:448]: Remove the increment. It was used before for debugging the stack variable and preventing it from being optimized away

File: phy/sx1276/sx1276_src/sx1276.c [hunk 6]
```diff
@@ -458,8 +457,8 @@ void initRadio(void)
     setSignalBandwidth(125E3);
     setCodingRate4(2);    
     //125kHz optimization
-    version = SX1276Read(REG_LR_DETECTOPTIMIZE);
-    version &= 0b01111111;
+    rad_version = SX1276Read(REG_LR_DETECTOPTIMIZE);
+    rad_version &= 0b01111111;
```
Comment (sramrajkar) [inline phy/sx1276/sx1276_src/sx1276.c:461]: Use new variable for this as it will over write the rad version variable

**Review by sramrajkar — APPROVED** [review]

**Pull request merged**